src/validation.cpp

```diff
@@ -1851,6 +1851,18 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // SEGWIT2X signalling.
+    if ( VersionBitsState(pindex->pprev, chainparams.GetConsensus(), Consensus::DEPLOYMENT_SEGWIT2X, versionbitscache) == THRESHOLD_ACTIVE &&
```

Comment:
> Will the current code start signaling bit 1 support as soon as bit 4 is locked in (rather than 2w later when it activates), and if not, shouldn't it?

I'm not *that* familiar with the codebase in general, but it seems to already set all defined BIP9 flags in the locked-in and started state for the default nVersion for getblocktemplate, so Core 0.13.1 and later should do that already; that code isn't touched by SegWit2x. It would be up to miners to update their custom nVersion, most likely manually. [See this code.](https://github.com/bitcoin/bitcoin/blob/1b708f2cf3e6e6a17c5b6dbf2909a2ed2e35755a/src/validation.cpp#L1436)

Quoting @jameshilliard ;

> It is quite trivial actually for pool operators to set a custom nVersion in their stratum servers, some software has it as a config variable and for others it's a 1 line code change. Any reasonably sized pool should have no problem doing so(and many already set it manually) smaller solo miners appear to already set the segwit bit for the most part as well.